packages/resilience/src/circuit-breaker.spec.ts

```diff
@@ -0,0 +1,243 @@
+      method: 'get',
+      baseURL: host,
+      url: 'ok'
+    };
+    const request = () => axios.request(requestConfig);
+    const context: HttpMiddlewareContext = {
+      requestConfig,
+      uri: host,
+      tenantId: 'tenant1'
+    };
+
+    await executeWithMiddleware<AxiosResponse, HttpMiddlewareContext>(
+      [circuitBreakerHttp()],
+      context,
+      request
+    );
+    await executeWithMiddleware<AxiosResponse, HttpMiddlewareContext>(
+      [circuitBreakerHttp()],
+      { ...context, tenantId: 'tenant2' },
+      request
+    );
+
+    expect(Object.keys(circuitBreakers)).toEqual([
+      `${host}::tenant1`,
+      `${host}::tenant2`
+    ]);
+  });
+
+  it('creates one circuit breaker for all requests against the same host', async () => {
+    nock(host, {})
+      .get(/path-1/)
+      .reply(200)
+      .get(/path-2/)
+      .reply(200);
+
+    const requestConfigPath1: AxiosRequestConfig = {
+      method: 'get',
+      baseURL: host,
+      url: 'path-1'
+    };
+    const requestConfigPath2: AxiosRequestConfig = {
+      method: 'get',
+      baseURL: host,
+      url: 'path-2'
+    };
+    const request = requestConfig => () => axios.request(requestConfig);
+    const context: (
+      requestConfig: AxiosRequestConfig
+    ) => HttpMiddlewareContext = requestConfig => ({
+      requestConfig,
+      uri: host,
+      tenantId: 'tenant1'
+    });
+
+    await executeWithMiddleware<AxiosResponse, HttpMiddlewareContext>(
+      [circuitBreakerHttp()],
+      context(requestConfigPath1),
+      request(requestConfigPath1)
+    );
+    await executeWithMiddleware<AxiosResponse, HttpMiddlewareContext>(
+      [circuitBreakerHttp()],
+      context(requestConfigPath2),
+      request(requestConfigPath2)
+    );
+
+    expect(Object.keys(circuitBreakers)).toEqual([`${host}::tenant1`]);
+  });
+
+  it('does not open breaker for 401 xsuaa failures', async () => {
+    const mock = nock(host, {})
+      .post(/oauth\/token/)
+      .times(10)
+      .reply(401);
+
+    const requestConfig: AxiosRequestConfig = {
+      method: 'post',
+      baseURL: host,
+      url: '/oauth/token',
+      data: {
+        client_id: 'client_id',
+        client_secret: 'wrong_client_secret',
+        grant_type: 'client_credentials'
+      }
+    };
+    const context: HttpMiddlewareContext = {
+      requestConfig,
+      uri: host,
+      tenantId: 'myTestTenant'
+    };
+    const request = () => axios.request(requestConfig);
+    let keepCalling = !mock.isDone();
+    while (keepCalling) {
+      await expect(
+        executeWithMiddleware<AxiosResponse, HttpMiddlewareContext>(
+          [circuitBreakerHttp()],
+          context,
+          request
+        )
+      ).rejects.toThrowError(/Request failed with status code 401/);
+      keepCalling = !mock.isDone();
+    }
+    expect(circuitBreakers[`${host}::myTestTenant`].opened).toBe(false);
+  });
+
+  it('works together with a timeout', async () => {
```

Comment:
[nth] You could move this test to the resilience.spec.ts @fwilhe introduced. This contains all the combined tests.

Reply:
Done